src/Text/LLVM/PP.hs

```diff
@@ -72,13 +70,26 @@ ppLLVM38 = withConfig Config { cfgLoadImplicitType = False
                              , cfgUseDILocation    = True
                              }
 
-checkConfig :: LLVM => (Config -> Bool) -> Bool
+checkConfig :: (?config :: Config) => (Config -> Bool) -> Bool
 checkConfig p = p ?config
 
 
+type Fmt a = (?config :: Config) => a -> Doc
+
+
+-- | The LLVMPretty class has instances for most AST elements.  It allows the
+-- conversion of an AST element (and its sub-elements) into a Doc assembly format
+-- by simply using the 'llvmPP' method rather than needing to explicitly invoke
+-- the specific pretty-printing function for that element.
+class LLVMPretty a where llvmPP :: Fmt a
+
+instance LLVMPretty Module where llvmPP = ppModule
+instance LLVMPretty Symbol where llvmPP = ppSymbol
```

Comment:
There's not much payoff to adding this class just yet, since you don't call `llvmPP` anywhere. This isn't to say that I'm opposed to adding it here, but I think it remains to be seen if the payoff will be worth it in a subsequent patch.